data-model/src/main/java/io/micronaut/data/model/naming/NamingStrategy.java

```diff
@@ -141,10 +141,14 @@ public interface NamingStrategy {
                     foreignAssociation = association;
                 }
             }
+            String assocName = association.getKind() == Relation.Kind.EMBEDDED ? association.getAnnotationMetadata().stringValue(MappedProperty.class).orElse(null) : null;
+            if (assocName == null) {
```

Comment:
You can have orElseGet and eliminate this if

Reply:
Done https://github.com/micronaut-projects/micronaut-data/pull/1859/commits/b7684188701f18127e5ee36f55590f74fa589452